### PR TITLE
Feature/mechanical obstacle 

### PR DESCRIPTION
**Pull request by Davidek776**
This pull requests focuses on introducing new type of obstacles which are high obstacle. It is also introducing triggers that can be used to activate and deactivate methods in classes that are implementing ISwitchable abstract class, that was also introduced in this PR. High obstacles can be controlled by triggers, what makes them mechanical obstacles. High obstacles in this PR have verticalMovement action.

## TLDR:

---

1. Trigger Prefab
2. High Obstacle Prefab
3. Trigger and high obstacle connection
4. Universal ISwitchable abstract class for VerticalMovementClass and other classes with the same purpose
5. Update scene to make the ground bigger and position trigggers and mechanical obstacles to scene

### CODE REVIEW

**Review by simonfrayer — APPROVED** [review]
🦍
